File: surf-api-bukkit/surf-api-bukkit-server/src/main/java/dev/slne/surf/surfapi/bukkit/server/listener/ListenerManager.java
package dev.slne.surf.surfapi.bukkit.server.listener;

import dev.slne.surf.surfapi.bukkit.server.BukkitMain;
import org.bukkit.Bukkit;
import org.jetbrains.annotations.ApiStatus;

/**
 * The ListenerManager class is responsible for registering and unregistering listeners in the
 * Bukkit plugin. It provides methods to register and unregister listeners for specific events. The
 * listener manager is instantiated with a reference to the main plugin class, BukkitMain.
 *
 * @see BukkitMain
 */
@ApiStatus.Internal
public class ListenerManager {

  private final BukkitMain plugin;

  /**
   * The ListenerManager class is responsible for registering and unregistering listeners in the
   * Bukkit plugin. It provides methods to register and unregister listeners for specific events.
   * The listener manager is instantiated with a reference to the main plugin class, BukkitMain.
   *
   * @see BukkitMain
   */
  public ListenerManager(BukkitMain plugin) {
    this.plugin = plugin;
  }

  /**
   * Registers all listeners.
   */
  public void registerListeners() {
    Bukkit.getMessenger().registerOutgoingPluginChannel(plugin, "BungeeCord");
  }

  /**
   * Unregisters all listeners.
   */
  public void unregisterListeners() {
    Bukkit.getMessenger().unregisterOutgoingPluginChannel(plugin, "BungeeCord");
  }
}

File: surf-api-bukkit/surf-api-bukkit-server/src/main/kotlin/dev/slne/surf/surfapi/bukkit/server/listener/ListenerManager.kt
package dev.slne.surf.surfapi.bukkit.server.listener

import dev.slne.surf.surfapi.bukkit.api.event.register
import dev.slne.surf.surfapi.bukkit.server.gui.view.GuiViewListener
import dev.slne.surf.surfapi.bukkit.server.impl.glow.GlowingListener
import dev.slne.surf.surfapi.bukkit.server.impl.visualizer.visualizer.VisualizerListener
import dev.slne.surf.surfapi.bukkit.server.plugin
import org.bukkit.Bukkit

object ListenerManager {
    /**
     * Registers all listeners.
     */
    fun registerListeners() {
        Bukkit.getMessenger().registerOutgoingPluginChannel(plugin, "BungeeCord")
        VisualizerListener.register()
        GlowingListener.register()
        GuiViewListener.register()
    }

    /**
     * Unregisters all listeners.
     */
    fun unregisterListeners() {
        Bukkit.getMessenger().unregisterOutgoingPluginChannel(plugin, "BungeeCord")
    }
}
